In [32]:
import numpy as np
import pandas as pd

In [33]:
df = pd.read_csv("roadaccidents\\uk_road_accident.csv")

In [34]:
df.describe()

Latitude      Longitude  Number_of_Casualties  Number_of_Vehicles
count  660654.000000  660653.000000         660679.000000       660679.000000
mean       52.553866      -1.431210              1.357040            1.831255
std         1.406922       1.383330              0.824847            0.715269
min        49.914430      -7.516225              1.000000            1.000000
25%        51.490690      -2.332291              1.000000            1.000000
50%        52.315641      -1.411667              1.000000            2.000000
75%        53.453452      -0.232869              1.000000            2.000000
max        60.757544       1.762010             68.000000           32.000000

In [37]:
df.isnull()

Index  Accident_Severity  Accident Date  Latitude  Light_Conditions  \
0       False              False          False     False             False   
1       False              False          False     False             False   
2       False              False          False     False             False   
3       False              False          False     False             False   
4       False              False          False     False             False   
...       ...                ...            ...       ...               ...   
660674  False              False          False     False             False   
660675  False              False          False     False             False   
660676  False              False          False     False             False   
660677  False              False          False     False             False   
660678  False              False          False     False             False   

        District Area  Longitude  Number_of_Casualties  Number_of_Vehicles  \
0               False      False                 False               False   
1               False      False                 False               False   
2               False      False                 False               False   
3               False      False                 False               False   
4               False      False                 False               False   
...               ...        ...                   ...                 ...   
660674          False      False                 False               False   
660675          False      False                 False               False   
660676          False      False                 False               False   
660677          False      False                 False               False   
660678          False      False                 False               False   

        Road_Surface_Conditions  Road_Type  Urban_or_Rural_Area  \
0                         False      False                False   
1                         False      False                False   
2                         False       True                False   
3                         False      False                False   
4                         False       True                False   
...                         ...        ...                  ...   
660674                    False      False                False   
660675                    False      False                False   
660676                    False      False                False   
660677                    False      False                False   
660678                    False      False                False   

        Weather_Conditions  Vehicle_Type  
0                    False         False  
1                    False         False  
2                     True         False  
3                    False         False  
4                     True         False  
...                    ...           ...  
660674               False         False  
660675               False         False  
660676               False         False  
660677               False         False  
660678               False         False  

[660679 rows x 14 columns]

In [38]:
df.isnull().sum()

Index                          0
Accident_Severity              0
Accident Date                  0
Latitude                      25
Light_Conditions               0
District Area                  0
Longitude                     26
Number_of_Casualties           0
Number_of_Vehicles             0
Road_Surface_Conditions      726
Road_Type                   4520
Urban_or_Rural_Area           15
Weather_Conditions         14128
Vehicle_Type                   0
dtype: int64

<h1>1. How many accidents are recorded in the data set?</h1>

In [39]:
df[['Accident_Severity']].count()

Accident_Severity    660679
dtype: int64

In [40]:
len(df)

660679

<p>This shows the total number of accidents that happened in the dataset that been given</p>

<h1>2. What is the severity of the accidents in the data set?</h1>

In [41]:
df['Accident_Severity'].value_counts()

Accident_Severity
Slight     563801
Serious     88217
Fatal        8661
Name: count, dtype: int64

<p>Most of the accidents are slightly severe, but still there is too many serious and fatal accidents that been given</p>

<h1>3. On which  district area have the highest numbers of road accidents? </h1>

In [42]:
df['District Area'].value_counts()

District Area
Birmingham            13491
Leeds                  8898
Manchester             6720
Bradford               6212
Sheffield              5710
                      ...  
Berwick-upon-Tweed      153
Teesdale                142
Shetland Islands        133
Orkney Islands          117
Clackmannanshire         91
Name: count, Length: 422, dtype: int64

<p>This shows that there's a lot of district that is prone for road accidents, but the highest one is Birmigham</p>

<h1>4. What is the average number of vehicles per accident?</h1>

In [43]:
df['Number_of_Vehicles'].mean()

np.float64(1.8312554205597575)

<p>This sums up that maybe around 2 vehicles are there per accidents</p>

<h1>5. What are the typical numbers of casualties in an accidents?</h1>

In [44]:
df['Number_of_Casualties'].mean()

np.float64(1.357040257068864)

In [45]:
df['Number_of_Casualties'].describe()

count    660679.000000
mean          1.357040
std           0.824847
min           1.000000
25%           1.000000
50%           1.000000
75%           1.000000
max          68.000000
Name: Number_of_Casualties, dtype: float64

<p>Most accidents involve 1 casualty only</p>

<h1>6. How many vehicles are usually involve in accidents?</h1>

In [46]:
df['Number_of_Vehicles'].describe()

count    660679.000000
mean          1.831255
std           0.715269
min           1.000000
25%           1.000000
50%           2.000000
75%           2.000000
max          32.000000
Name: Number_of_Vehicles, dtype: float64

<p>Usually 2 vehicles are involve around</p>

<h1>7. Do accidents happen more in daylight or in darkness?</h1>

In [47]:
df['Light_Conditions'].value_counts()

Light_Conditions
Daylight                       484880
Darkness - lights lit          129335
Darkness - no lighting          37437
Darkness - lighting unknown      6484
Darkness - lights unlit          2543
Name: count, dtype: int64

<p>This shows that more accidents happen in daylight, tho dark unlit conditions show heightened severity</p>

<h1>8. Which surface type correspond with more frequent accidents?</h1>

In [48]:
df['Road_Surface_Conditions'].value_counts()

Road_Surface_Conditions
Dry                     447821
Wet or damp             186708
Frost or ice             18517
Snow                      5890
Flood over 3cm. deep      1017
Name: count, dtype: int64

<p>This simply shows that dry surface dominate more accidents than other type of surface</p>

<h1>9. Does  accident happened more on urban or rural area?</h1>

In [49]:
df['Urban_or_Rural_Area'].describe()

count     660664
unique         3
top        Urban
freq      421663
Name: Urban_or_Rural_Area, dtype: object

In [50]:
df['Urban_or_Rural_Area'].value_counts()

Urban_or_Rural_Area
Urban          421663
Rural          238990
Unallocated        11
Name: count, dtype: int64

<p>It simply shows that the Urban are prone to more accidents, likely due to higher density of vehicles and people around</p>

<h1>10. Which weather is more likely prone to accidents?</h1>

In [51]:
df['Weather_Conditions'].value_counts()

Weather_Conditions
Fine no high winds       520885
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<p>Surprisingly Clear weather are more prone to accidents</p>

<h1>11. How many accidents happened in snowing with no high winds?</h1>

In [54]:
df['Weather_Conditions'].value_counts()

Weather_Conditions
Fine no high winds       520885
Raining no high winds     79696
Other                     17150
Raining + high winds       9615
Fine + high winds          8554
Snowing no high winds      6238
Fog or mist                3528
Snowing + high winds        885
Name: count, dtype: int64

<p>This shows that accidents happened in snowing with no high winds are 6238 which is low than the other weathers</p>

<h1>12. Which light settings correspond with the most severe outcomes?</h1>

In [60]:
pd.crosstab( df['Light_Conditions'], df['Accident_Severity'])

Accident_Severity            Fatal  Serious  Slight
Light_Conditions                                   
Darkness - lighting unknown     68      794    5622
Darkness - lights lit         1860    19130  108345
Darkness - lights unlit         45      360    2138
Darkness - no lighting        1612     7174   28651
Daylight                      5076    60759  419045

<p>Dark roads without lighting shows severe and fatal accident counts</p>

<h1>13. Which latitudes recorded the most accidents?</h1>

In [61]:
df['Latitude'].value_counts()

Latitude
52.458798    50
52.949719    45
53.083165    33
52.967634    31
52.938860    29
             ..
56.880265     1
56.865367     1
57.528413     1
57.194499     1
51.493490     1
Name: count, Length: 511618, dtype: int64

In [62]:
df['Latitude'].count()

np.int64(660654)

<h1>14. Which longitudes recorded the most accidents?</h1>

In [65]:
df['Longitude'].value_counts()

Longitude
-0.977611    45
-1.871043    36
-0.816789    33
-1.190861    31
-1.216694    29
             ..
-4.169644     1
-3.798648     1
-3.630574     1
-3.836563     1
-3.924477     1
Name: count, Length: 529766, dtype: int64

In [66]:
df['Longitude'].count()

np.int64(660653)

<h1>15. Where do the latitude and longitude values fall within the data sets?</h1>

In [67]:
df[['Latitude', 'Longitude']].describe()

Latitude      Longitude
count  660654.000000  660653.000000
mean       52.553866      -1.431210
std         1.406922       1.383330
min        49.914430      -7.516225
25%        51.490690      -2.332291
50%        52.315641      -1.411667
75%        53.453452      -0.232869
max        60.757544       1.762010

<p>This confirms that the spans of the data set cover both northern/sothern and eastern/western regions</p>

<h1>16. Is there a measurable connections between vehicle counts and casualties?</h1>

In [68]:
df[['Number_of_Vehicles', 'Number_of_Casualties']].corr()

Number_of_Vehicles  Number_of_Casualties
Number_of_Vehicles              1.000000              0.228889
Number_of_Casualties            0.228889              1.000000

<p> This simply shows that there is connection, which indicates that more vehicles are prone to more casualties as well</p>

<h1>17. Which road design contributes most to accident frequency?</h1>

In [69]:
df["Road_Type"].value_counts()

Road_Type
Single carriageway    492143
Dual carriageway       99424
Roundabout             43992
One way street         13559
Slip road               7041
Name: count, dtype: int64

<p>This data shows that Single carriageway are the most common sites for accidents</p>

<h1>18. Does the weather contributes to the severe damage of every accidents?</h1>

In [71]:
pd.crosstab(df['Weather_Conditions'], df['Accident_Severity'])

Accident_Severity      Fatal  Serious  Slight
Weather_Conditions                           
Fine + high winds        175     1245    7134
Fine no high winds      7100    72046  441739
Fog or mist               82      483    2963
Other                    165     1801   15184
Raining + high winds     145     1261    8209
Raining no high winds    848     9468   69380
Snowing + high winds       3      109     773
Snowing no high winds     36      565    5637

<p>It shows that weathers contributes, it simply shows that a severe outcomes are more frequent in poor weather</p>

<h1>19. To what extend does road surface influence road severity?</h1>

In [73]:
pd.crosstab(df['Road_Surface_Conditions'], df['Accident_Severity'])

Accident_Severity        Fatal  Serious  Slight
Road_Surface_Conditions                        
Dry                       5788    61638  380395
Flood over 3cm. deep        23      152     842
Frost or ice               193     2007   16317
Snow                        35      565    5290
Wet or damp               2620    23785  160303

<p>This demonstrate that wet and icy have a higher risk of severe accidents</p>

<h1>20. What road type has the most road severity?</h1>

In [74]:
pd.crosstab(df["Road_Type"], df['Accident_Severity'])

Accident_Severity   Fatal  Serious  Slight
Road_Type                                 
Dual carriageway     1815    11746   85863
One way street         95     1655   11809
Roundabout            142     3665   40185
Single carriageway   6527    70059  415557
Slip road              49      611    6381

<p>The data set shows that most of the road severity comes from dual carriageway</p>